### PR TITLE
Add /api/v1/contracts/call to open API spec

### LINKED ISSUE #5584: Add `/api/v1/contracts/call` to open API spec
### Problem

The open API spec does not have new API listed `/api/v1/contracts/call`.

https://testnet.mirrornode.hedera.com/api/v1/docs/#/

### Solution

Update the open API spec. 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

Add /api/v1/contracts/call to open API spec

**Related issue(s)**:

Fixes #5584 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 1]
```diff
@@ -296,6 +296,36 @@ paths:
           $ref: '#/components/responses/NotFoundError'
       tags:
         - blocks
+
+  /api/v1/contracts/call:
+      post:
+        summary: eth_call and eth_estimateGas
+        description: Returns a result from execution of eth_call and eth_estimateGas. If `estimate` field set to true eth_estimateGas is executed
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:303]: Let's not use eth_call or eth_estimateGas terminology. This is not a JSON-RPC API, it is a REST API that calls a contract. So just generic Hedera terms focused on contract calls or EVM execution.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 2]
```diff
@@ -296,6 +296,36 @@ paths:
...
+        requestBody:
+          required: true
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  to:
+                    type: string
+                  data:
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:315]: nit: sort

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 3]
```diff
@@ -296,6 +296,36 @@ paths:
...
+          - url:
+        requestBody:
+          required: true
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  to:
+                    type: string
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:314]: Missing description for all request and response fields.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 4]
```diff
@@ -296,6 +296,36 @@ paths:
...
+        description: Returns a result from execution of eth_call and eth_estimateGas. If `estimate` field set to true eth_estimateGas is executed
+        servers:
+          - url:
+        requestBody:
+          required: true
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:312]: Missing many fields

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 5]
```diff
@@ -296,6 +296,36 @@ paths:
...
+                  to:
+                    type: string
+                  data:
+                    type: string
+        responses:
+          200:
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:318]: Add error response codes and format. need to add detail and data to generic error response structure.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 6]
```diff
@@ -296,6 +296,106 @@ paths:
...
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/Web3Error'
+          429:
+            description: Too many requests
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/Web3Error'
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-rest/api/v1/openapi.yml:396]: We have also `501 Not Implemented` exception. We can also sort the http errors in ascending order.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 7]
```diff
@@ -296,6 +296,106 @@ paths:
...
+                    maxLength: 40
+                    example: "00000000000000000000000000000000000003f4"
+                  value:
+                    format: int64
+                    minimum: 0
+                    type: integer
+                    description: Value sent with this transaction
+                    example: 0
+                required:
+                  - to
```
Comment (xin-hedera) [inline hedera-mirror-rest/api/v1/openapi.yml:361]: is `to` the only required property?

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 8]
```diff
@@ -296,6 +296,106 @@ paths:
...
+                properties:
+                  block:
+                    type: string
+                    description: Hexadecimal block number or the string "latest"
+                    example: "latest"
+                  data:
+                    type: string
+                    description: Hexadecimal method signature and encoded parameters.
+                    format: binary
+                    maxLength: 6144 * 2
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-rest/api/v1/openapi.yml:320]: No, the code has 6144x2.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 9]
```diff
@@ -296,6 +296,106 @@ paths:
...
+                    example: "0x0198489200000000000000000000000000000000000000000000000000000000000003ee"
+                  estimate:
+                    type: boolean
+                    description: Whether gas estimation is called
+                    example: true
+                  from:
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:326]: Add nullable: true

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 10]
```diff
@@ -296,6 +296,106 @@ paths:
...
+                    type: string
+                    pattern: '^(0x)?[A-Fa-f0-9]{40}$'
+                    description: 20 Bytes - The hexed address the transaction is sent from.
+                    format: binary
+                    minLength: 40
+                    maxLength: 40
+                    example: "00000000000000000000000000000000000004e2"
+                  gas:
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:334]: Add nullable: true. Add default of 120000000 to description.

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 6 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/api/v1/openapi.yml:396]: Yes, put all current response codes and adjust them later.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 11]
```diff
@@ -2225,6 +2325,24 @@ components:
...
+              type: array
+              items:
+                type: object
+                properties:
+                  message:
+                    type: string
+                  detail:
+                    type: string
+                  data:
+                    type: string
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:2344]: These are missing description, example, pattern, nullable: false, format. Also should sort.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — DISMISSED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**